### ISSUE TITLE
Uncaught Error: Option 'schema' can not be changed

### ISSUE THREAD

**Issue by laszbalo** [tgecho/react-prosemirror#10]
If I create a react-prosemirror component instance with a custom schema, then any time the parent component' render method runs, I get the above error.

Just a side note, inside componentDidUpdate, when previous and current options are compared, the non-primitive values (schema, place, commands and doc [ if it is in some fancy format]) will be never  equal, due to the 'shallow' equality checking, therefore pm.setOption is called unnecessary. I have not dig into this deeper, so not sure how does this affect performance, however it might worth to keep it in mind.

So back to the real issue, since I am not changing the schema in the parent component, I just made a dirty hack in componentDidUpdate, and filtered out schema, like this:

``` javascript
Object.keys(current).filter( k => k!=='schema').forEach(...);
```

However if someone wants pass in different schema objects to react-prosemirror, based on some state or props of the parent component, then the inability of changing schema of an existing ProseMirror instance, could be problematic. What do you think?

**Comment by tgecho** [comment]
Strange. I don't see a good way to check if an option can be updated from the outside. I think we're just going to need to depend on ProseMirror's error message (which is pretty reasonable).

Are you creating the schema inside the render function of the component?

```
const MyComponent = React.createClass({
    render() {
        const options = {schema: new Schema(schemaSpec)}
        return <ProseMirror options={options} />
    }
})
```

If you do that, the schema will be recreated every time and always fail the strict/shallow equality check. Instead, maybe try creating it outside of the component (or once, in componentWillMount)?

```
const mySchema = new Schema(schemaSpec)
const MyComponent = React.createClass({
    render() {
        const options = {schema: mySchema}
        return <ProseMirror options={options} />
    }
})
```

**Comment by tgecho** [comment]
I think any scenarios that require changing the schema dynamically are too tricky to try to handle automatically. I suppose you could pull the content out and recreate the ProseMirror instance, but that doesn't sound like something that's safe enough to do at the level of this library.

**Comment by tgecho** [comment]
I played with this a moment and I confirmed that at least the scenarios I described are correct. I don't know about your specific case.

Allowing the exception to bubble up was scrambling the editor unpleasantly, so I just pushed a change that catches it and logs the original error along with a hopefully helpful warning. https://github.com/tgecho/react-prosemirror/commit/54657745e06253e1b436bb5f6c6178e702e0486a
